docs/text-predicates.rst

```diff
@@ -0,0 +1,215 @@
+----------------------------
+
+The Text Predicates are broadly divided into 2 categories as:
+    - Text search predicates which match against the individual words inside a text string after it has been tokenized.
+        These predicates are not case sensitive.
+
+        - TextContains_
+        - TextContainsFuzzy_
+        - TextContainsPrefix_
+        - TextContainsRegex_
+
+    - String search predicates which match against the entire string value
+
+        - TextFuzzy_
+        - TextPrefix_
+        - TextRegex_
+
+.. note::
+    All the bellow examples are based on Graph of the Gods graph which comes pre-packaged with JanusGraph
+
+^^^^^^^^^^^^^^^
+TextContains
+^^^^^^^^^^^^^^^
+
+This Predicate is part of *Text* package.
+TextContains is used to query for partial matches in values of properties.
+The query is to find all edges where ``reason`` contains ``breezes``
+
+
+.. code-block:: python
+
+    from janusgraph_python.core.attribute.text import Text
+
+    # Find all edges which contains the word breezes
+    edges = g.E().has("reason", Text.textContains("breezes")).next()
+    print(edges)
+
+    ==> e[55m-6hc-b2t-3bk][8400-lives->4304]
+
+    # Find all edges which contains the word breezes and loves
+    edges = g.E().has("reason", Text.textContains("breezes")).has("reason", Text.textContains("loves")).valueMap(True).next()
+    print(edges)
+
+    ==> {<T.id: 1>: '2dj-37c-9hx-360', 'reason': 'loves fresh breezes', <T.label: 3>: 'lives'}
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+TextContainsFuzzy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This Predicate is part of *Text* package.
+TextContainsFuzzy is used to query for partial matches with Fuzzy logic in values of properties.
+We know there is a vertex with *hercules* value on property `name`. We query all vertices with value as *hercauleas*
+Similarly, we use in Conjugation with multiple other attributes to query edge as in above example.
+
+
+.. code-block:: python
+
+    from janusgraph_python.core.attribute.text import Text
+
+    hercules = g.V().has("name", Text.textContainsFuzzy("hercauleas")).valueMap(True).next()
+    print(hercules)
+
+    ==> {'name': ['hercules'], <T.id: 1>: 4312, <T.label: 3>: 'demigod', 'age': [30]}
+
+    edges = g.E().has("reason", Text.textContainsFuzzy("breezs")).has("reason", Text.textContains("loves")).valueMap(True).next()
+    print(edges)
+
+    ==> {<T.id: 1>: '2dj-37c-9hx-360', 'reason': 'loves fresh breezes', <T.label: 3>: 'lives'}
+
+    edges = g.E().has("reason", Text.textContainsFuzzy("breezs")).has("reason", Text.textContainsFuzzy("luves")).valueMap(True).next()
+    print(edges)
+
+    ==> {<T.id: 1>: '2dj-37c-9hx-360', 'reason': 'loves fresh breezes', <T.label: 3>: 'lives'}
+
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+TextContainsPrefix
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This Predicate is part of *Text* package.
+TextContainsPrefix is used to query for where the value
+(if sentence, it is segmented, & any word in segment is considered)
```

Comment:
This isn't specific to `TextContainsPrefix` so I wouldn't explain it here. Otherwise, why isn't this also explained for the other `TextContains_` predicates?